lib/facter/letsencrypt_directory.rb

```diff
@@ -0,0 +1,29 @@
+require 'openssl'
+require 'pathname'
+
+Facter.add(:letsencrypt_directory) do
+  setcode do
+    certs = {}
+
+    # locate the certificate repository
+    livedir = [
+      '/etc/letsencrypt/live',
+      '/etc/certbot/live'
+    ].map { |path| Pathname.new path }.find(&:directory?)
+
+    # collect the sans from each certificate
+    if livedir
+      Pathname.new(livedir).children.select(&:directory?).each do |path|
+        pem = File.join(path, 'cert.pem')
+        cert = OpenSSL::X509::Certificate.new(File.new(pem).read)
+        san = cert.extensions.find { |e| e.oid == 'subjectAltName' }
+        names = san.value.split(',').map { |entry| entry.split(':')[1] }
+        names.each do |n|
+          certs[n] = path.to_s
+        end
+      end
+    end
+
+    certs
```

Comment:
should we instead use a structured fact? Would make the function obsolete or easier

Reply:
Amending git history is a PITA. Unless there's some reason it matters, I'd rather just leave the email as-is. 

The fact is structured (it's a hash). While you can certainly index into it directly, the function also handles wildcards, which isn't something a structure is going to provide.